keras/ops/core_test.py

```diff
@@ -215,6 +241,53 @@ def test_slice_update(self):
         outputs = core.slice_update(inputs, start_indices, updates)
         self.assertAllClose(outputs[1:3, 1:3, 2:4, 2:4], np.zeros([2, 2, 2, 2]))
 
+    def test_scan(self):
+        def f(carry, x):
+            if type(carry) is list or type(carry) is tuple:
+                x += carry[0]
+                carry = carry[0]
+            else:
+
+                x += carry
+            return carry, x
+
+        init_carr = np.array(1)
+        xs = np.array([0, 1, 2, 3, 4, 5, 6])
+
+        carry_op, ys_op = core.scan(
+            f, init_carr, xs, length=len(xs), reverse=False
+        )
+        carry_jax, ys_jax = jax.lax.scan(
+            f, init_carr, xs, length=len(xs), reverse=False
+        )
+
+        ys_op = ys_op.tolist()
+        ys_jax = ys_jax.tolist()
+
+        self.assertEqual(carry_op, carry_jax)
+        self.assertListEqual(ys_op, ys_jax)
+
+        init_carr = np.array(1.1)
+        xs = np.array([0.1, 1.2, 2.3, 3.4, 4.5, 5.6, 6.7])
+
+        carry_op, ys_op = core.scan(
+            f, init_carr, xs, length=len(xs), reverse=True
+        )
+        carry_jax, ys_jax = jax.lax.scan(
+            f, init_carr, xs, length=len(xs), reverse=True
+        )
+
+        ys_op = ys_op.tolist()
+        ys_jax = ys_jax.tolist()
+
+        self.assertEqual(carry_op, carry_jax)
+
+        for i in range(len(ys_op)):
+            ys_op[i] = round(ys_op[0],2)
+            ys_jax[i] = round(ys_jax[0], 2)
+
+        self.assertListEqual(ys_op, ys_jax)
```

Comment:
Use `assertAllClose` for numerical checks.

Please test more input argument configurations. You could use a parameterize test for that.

Reply:
New test added. I couldn't use parameterize because I can't pass arguments.I used instead for cycle.